### ISSUE TITLE
runstaskell seems to not use sandbox

### ISSUE THREAD

**Issue by jkarni** [soenkehahn/runstaskell#5]
```
$ runstaskell --bootstrap rc-1.15                                                                 
Writing a default package environment file to
/home/jkarni/Documents/Zalora/runstaskell/.cabal-sandbox/share/x86_64-linux-ghc-7.8.4/runstaskell-0.1/sandboxes/rc-1.15/cabal.sandbox.config
Using an existing sandbox located at
/home/jkarni/Documents/Zalora/runstaskell/.cabal-sandbox/share/x86_64-linux-ghc-7.8.4/runstaskell-0.1/sandboxes/rc-1.15/.cabal-sandbox
--2015-04-19 14:12:11--  http://www.stackage.org/snapshot/lts-1.15/cabal.config
Resolving www.stackage.org (www.stackage.org)... 54.236.93.122, 54.236.141.227
Connecting to www.stackage.org (www.stackage.org)|54.236.93.122|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/plain]
Saving to: ‘/home/jkarni/Documents/Zalora/runstaskell/.cabal-sandbox/share/x86_64-linux-ghc-7.8.4/runstaskell-0.1/sandboxes/rc-1.15/cabal.config’

    [  <=>                                                                                 ] 30.440       126KB/s   in 0,2s   

2015-04-19 14:12:11 (126 KB/s) - ‘/home/jkarni/Documents/Zalora/runstaskell/.cabal-sandbox/share/x86_64-linux-ghc-7.8.4/runstaskell-0.1/sandboxes/rc-1.15/cabal.config’ saved [30440]

Resolving dependencies...
In order, the following would be installed:
ansi-wl-pprint-0.6.7.2 (new package)
base64-bytestring-1.0.0.1 (new package)
cereal-0.4.1.1 (new package)
clock-0.4.1.3 (new package)
cmdargs-0.10.12 (new package)
directory-tree-0.12.0 (new package)
dlist-0.7.1.1 (new version)
data-default-instances-dlist-0.0.1 (reinstall) changes: dlist-0.7.1 -> 0.7.1.1
data-default-0.5.3 (reinstall)
errorcall-eq-instance-0.1.0 (latest: 0.2.0.1) (new package)
base-compat-0.5.0 (latest: 0.7.1) (new version)
foreign-var-0.0.0.1 (latest: 0.1) (new package)
mmorph-1.0.4 (new package)
prelude-extras-0.4 (new package)
primitive-0.5.4.0 (latest: 0.6) (new version)
random-1.0.1.1 (latest: 1.1) (new version)
MonadRandom-0.3.0.1 (latest: 0.3.0.2) (new package)
safe-0.3.8 (new package)
syb-0.4.4 (new version)
tagged-0.7.3 (latest: 0.8.0.1) (new package)
text-1.2.0.4 -integer-simple (new version)
blaze-builder-0.3.3.4 (latest: 0.4.0.1) (new package)
cookie-0.4.1.4 (new package)
hashable-1.2.3.2 (new version)
case-insensitive-1.2.0.4 (new package)
http-types-0.8.6 (new package)
mime-mail-0.4.8.2 (new package)
mime-types-0.1.0.6 (new package)
nats-1 (new package)
parsec-3.1.8 (latest: 3.1.9) (new version)
network-uri-2.6.0.1 (new package)
publicsuffixlist-0.1 (new package)
scientific-0.3.3.8 (new version)
attoparsec-0.12.1.5 (latest: 0.12.1.6) (new version)
string-conversions-0.3.0.3 (new package)
tf-random-0.5 (reinstall) changes: primitive-0.6 -> 0.5.4.0, random-1.1 ->
1.0.1.1
QuickCheck-2.7.6 (latest: 2.8.1) (new version)
quickcheck-io-0.1.1 (reinstall) changes: QuickCheck-2.8.1 -> 2.7.6
hspec-core-2.1.5 (reinstall) changes: QuickCheck-2.8.1 -> 2.7.6, random-1.1 ->
1.0.1.1
hspec-2.1.5 (reinstall) changes: QuickCheck-2.8.1 -> 2.7.6
transformers-compat-0.3.3.3 (latest: 0.4.0.4) (new package)
distributive-0.4.4 (new package)
optparse-applicative-0.11.0.2 (new package)
transformers-base-0.4.4 (new package)
monad-control-0.3.3.1 (latest: 1.0.0.4) (new package)
lifted-base-0.2.3.6 (new package)
enclosed-exceptions-1.0.1.1 (new package)
resourcet-1.1.4.1 (new package)
unordered-containers-0.2.5.1 (new package)
semigroups-0.16.2.2 (new package)
vector-0.10.12.3 (new package)
aeson-0.8.0.2 +old-locale (new package)
cassava-0.4.2.2 (latest: 0.4.2.3) (new package)
void-0.7 (new package)
conduit-1.2.4 (new package)
contravariant-1.2.2.1 (latest: 1.3.1) (new package)
comonad-4.2.5 (new package)
semigroupoids-4.2 (latest: 4.3) (new package)
bifunctors-4.2.1 (new package)
profunctors-4.3.2 (latest: 4.4.1) (new package)
free-4.10.0.1 (latest: 4.11) (new package)
either-4.3.3.2 (new package)
errors-1.4.7 (new package)
yaml-0.8.10.1 (latest: 0.8.11) (new package)
zlib-0.5.4.2 (latest: 0.6.0.0) (new package)
streaming-commons-0.1.10.0 (latest: 0.1.12) (new package)
http-client-0.4.9 (latest: 0.4.11.1) (new package)
cabal: The following packages are likely to be broken by the reinstalls:
libmpd-0.9.0.1
X11-1.6.1.2
xmonad-extras-0.12
xmonad-contrib-0.11.3
xmonad-0.11
X11-xft-0.3.1
QuickCheck-2.8.1
Use --force-reinstalls if you want to install anyway.
runstaskell: callCommand: cabal install aeson base-compat case-insensitive cassava cmdArgs containers deepseq directory directory-tree either enclosed-exceptions errors filepath http-client mime-mail mtl optparse-applicative process safe silently string-conversions tagged temporary transformers yaml hspec (exit 1): failed
```

The packages that would be broken are global packages. The `sandboxes/rc-1.15` folder does get created, and the `cabal.config` and `cabal.sandbox.config` files are generated, and point to the right locations. 

Using `cabal exec runstaskell ...` results in the same error.

**Comment by soenkehahn** [comment]
I tried to reproduce this but without success. I tried installing `runstaskell` in a sandbox and put packages that would be overwritten into my user database (`~/.ghc`). Then I started `./.cabal-sandbox/bin/runstaskell --bootstrap rc-1.15`, but that succeeded. @jkarni: Any idea what I could do differenly to reproduce the bug?

**Comment by soenkehahn** [comment]
We've ran into this on @tuleism's machine. The problem is that the packages that `runstaskell` wants to install into the sandbox conflict with packages in `ghc`'s global package database (in e.g. `/usr/lib`).

This is not a problem specific to `runstaskell` but to cabal sandboxes. I don't know a solution.

**Comment by soenkehahn** [comment]
This answer recommends using `--force-reinstalls`, which should be safe: http://stackoverflow.com/questions/21587200/how-to-get-cabal-to-ignore-the-global-package-db-when-using-a-sandbox

**Comment by soenkehahn** [comment]
Some context:

If you have packages installed globally, unfortunately they will be available to `runstaskell`. So you will be able to use these packages although other users of `runstaskell` will not necessarily have those.

With @tuleism's PR it may happen that those additional packages are broken by the packages in your `runstaskell` sandbox (broken only when the sandbox is used, that is).

**Comment by soenkehahn** [comment]
Fixed through #6.
